lib/services/platform-service.ts

```diff
@@ -570,19 +571,31 @@ export class PlatformService implements IPlatformService {
 		this.$logger.info(`Copied file '${packageFile}' to '${targetPath}'.`);
 	}
 
-	public removePlatforms(platforms: string[]): void {
+	public async removePlatforms(platforms: string[]): Promise<void> {
 		this.$projectDataService.initialize(this.$projectData.projectDir);
 
-		_.each(platforms, platform => {
+		for (let platform of platforms) {
 			this.validatePlatformInstalled(platform);
 			let platformData = this.$platformsData.getPlatformData(platform);
+			if (platform.toLowerCase() === "android") {
+				await this.gradleStop(platformData.projectRoot);
+			}
 
 			let platformDir = path.join(this.$projectData.platformsDir, platform);
 			this.$fs.deleteDirectory(platformDir);
 			this.$projectDataService.removeProperty(platformData.frameworkPackageName);
 
 			this.$logger.out(`Platform ${platform} successfully removed.`);
-		});
+		}
+	}
+
+	private async gradleStop(projectRoot: string): Promise<ISpawnResult> {
```

Comment:
looks great, just move logic to `android-project-service.ts`, and call through `platformData.platformProjectService...`